tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@xiang90 @heyitsanthony : its confirmed now that its not a bug. The behavior which @heyitsanthony mentioned is correct. I moved out the rc.Close() after cancel and all channels got closed.
Will update the PR with the changes.
Thanks!